vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSlideController.java

```diff
@@ -79,21 +85,33 @@ public String slide(Model model, @PathVariable("slideId") String slideId, @PathV
         model.addAttribute("firstSlide", module.getStartSequence().getSlides().get(0).getId());
         String nextSlideId = "";
         String prevSlideId = "";
-        int slideIndex = 0;
-        slideIndex = sequenceSlides.indexOf(sildeManager.getSlide(slideId));
+        ISlide currentSlide = sildeManager.getSlide(slideId);
+        int slideIndex = sequenceSlides.indexOf(currentSlide);
         int slideSize = sequenceSlides.size();
         if (slideSize > slideIndex + 1) {
             nextSlideId = sequenceSlides.get(slideIndex + 1).getId();
         }
         if (slideIndex > 0) {
             prevSlideId = sequenceSlides.get(slideIndex - 1).getId();
         }
+        model.addAttribute("sequences",moduleManager.getModuleSequences(moduleId));
         model.addAttribute("sequence",sequenceExist);
         model.addAttribute("slides", sequenceSlides);
         model.addAttribute("currentSequenceId", sequenceId);
         model.addAttribute("nextSlide", nextSlideId);
         model.addAttribute("prevSlide", prevSlideId);
-        model.addAttribute("currentSlideCon", sildeManager.getSlide(slideId));
+        model.addAttribute("currentSlideCon", currentSlide);
+        if(currentSlide instanceof BranchingPoint) {
+            model.addAttribute("choices", ((BranchingPoint)currentSlide).getChoices());
+            if(back) {
+                choiceHistory.removeLastElementFromSequenceSlideHistory();
```

Comment:
you might want to make sure that branching point and sequence are the same as the current slide ones. otherwise, if someone just attached the back parameter to a random branching point or it gets accidentally passed to it, this would cause the history to get out of sync.